analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -359,9 +359,9 @@ private DayActivity createNewDayActivity(UserAnonymized userAnonymizedEntity, Ac
 		{
 			weekActivity = WeekActivity.createInstance(userAnonymizedEntity, matchingGoal, startOfWeek.getZone(),
 					startOfWeek.toLocalDate());
+			matchingGoal.addWeekActivity(weekActivity);
 		}
 		weekActivity.addDayActivity(dayActivity);
-		matchingGoal.addWeekActivity(weekActivity);
```

Comment:
Found this little bug through the unit test. In practice, it currently doesn't do any harm, as the foreign key relationship causes the entity to be present just once.